File: guests/swift/commands.swift
// The commands scene, Swift port — guests/rust/commands.rs,
// tools/scenes/commands.steps.

import Foundation

let app = KayaApp()
var settingsCount = 0

app.build { tx in
    let status = tx.signal(.str("ready"))
    let details = tx.signal(.bool(false))
    let sort = tx.signal(.f64(0.0))

    // Reload keeps this menu non-empty once macOS moves Settings out.
    let file = tx.menu(
        "File",
        items: [
            tx.item("Reload"),
            tx.item("Settings…", shortcut: "primary+comma", role: KayaAppTx.roleSettings) { t in
                // Fires twice on purpose: the chord and the declared path.
                settingsCount += 1
                t.write(status, .str("settings \(settingsCount)"))
            },
        ])

    // Option order IS the index vocabulary: Name = 0, Date = 1.
    let view = tx.menu(
        "View",
        items: [
            tx.toggle("Details", checked: details, shortcut: "primary+backslash") { t, on in
                t.write(status, .str(on ? "details on" : "details off"))
            },
            tx.radioGroup(
                "Sort",
                options: [
                    tx.option("Name", shortcut: "primary+1"),
                    tx.option("Date", shortcut: "primary+2"),
                ], value: sort
            ) { t, index in
                t.write(status, .str(index == 1 ? "sorted date" : "sorted name"))
            },
        ])
    tx.window(title: "commands", menus: [file, view])

    let root = tx.column {
        tx.label(bind: status)  // label#0
    }
    tx.mount(root)
}

app.run()

File: guests/ocaml/commands.ml
(* The commands scene, OCaml port — guests/rust/commands.rs,
   tools/scenes/commands.steps. *)

open Kaya_app

let () =
  let app = Kaya_app.create () in
  let settings_count = ref 0 in

  build app (fun () ->
     let status = signal Scalar.Str ("ready") in
     let details = signal Scalar.Bool (false) in
     let sort = signal Scalar.F64 (0.0) in

     window ~title:"commands"
       ~menus:
         [
           (* Reload keeps this menu non-empty once macOS moves Settings out. *)
           menu ~label:"File"
             [
               item ~label:"Reload";
               item ~label:"Settings…" ~shortcut:"primary+comma"
                 ~role:Menu_role.Settings
                 ~on_activate:(fun () ->
                   (* Fires twice on purpose: the chord and the declared
                      path. *)
                   incr settings_count;
                   write status ((Printf.sprintf "settings %d" !settings_count)));
             ];
           (* Option order IS the index vocabulary: Name = 0, Date = 1. *)
           menu ~label:"View"
             [
               toggle ~label:"Details" ~bind_checked:details
                 ~shortcut:"primary+backslash"
                 ~on_toggle:(fun on ->
                   write status
                     (if on then "details on" else "details off"));
               radio_group ~label:"Sort" ~bind_value:sort
                 ~on_select:(fun index ->
                   write status
                     (if index = 1 then "sorted date" else "sorted name"))
                 [
                   option ~label:"Name" ~shortcut:"primary+1";
                   option ~label:"Date" ~shortcut:"primary+2";
                 ];
             ];
         ]
       ();

     let root = column [ label ~bind:status (* label#0 *) ] () in
     mount root);

  exit (run app)
